src/libraries/System.Drawing.Common/src/System/Drawing/GdiplusNative.Unix.cs

```diff
@@ -62,6 +63,9 @@ internal static IntPtr LoadNativeLibrary()
 
             private static void PlatformInitialize()
             {
+                if (!LocalAppContextSwitches.EnableUnixSupport)
+                    throw new PlatformNotSupportedException(SR.PlatformNotSupported_Unix);
+
                 LibraryResolver.EnsureRegistered();
```

Comment:
Should the check be inside `EnsureRegistered` so that it covers the other place where `EnsureRegistered` is called from as well?

Reply:
What other places could call these? I see this being called when loading libcups, do we want that to trigger this exception when we register to load `libcups` as well?

Reply:
I think we want to throw any time it does something non-trivial (ie PInvokes into unmanaged library). I am not sure whether there is path to hit libcups without hitting libgdiplus.

Reply:
I'm not sure either, but I think your suggestion makes sense. I'll move it to `EnsureRegistered`.

Reply:
Done, thanks.